### ISSUE TITLE
ColumnHolders do not give expected errors when values fall out of Java type bounds

### ISSUE THREAD

**Issue by jjbrosnan** [deephaven/deephaven-core#832]
**Description**

ColumnHolder constructor methods (byteCol, intCol, etc.) do not give errors when they are supplied with values out of bounds of the corresponding Java type.

**Steps to reproduce**

Start up the IDE in Python (or Groovy if you want, just exclude the import statements).  Both of the code blocks below cause this behavior:

from deephaven.TableTools import newTable, byteCol
table = newTable(
    byteCol("A", 128)
)

from deephaven.TableTools import newTable, shortCol
table = newTable(
    shortCol("A", 100000000)
)

**Expected results**

An error with a stack trace telling me a byte can't support numbers larger than 127 or less than -128.  Same for shorts with the appropriate bounds.  Note that if you give another data type (decimal, string, or char) that an error is generated as expected.

**Actual results**

An empty table with a spinning deephaven logo.  No error, no stack trace in the logs.  It makes it look like the IDE "broke", but you can still use the IDE as normal.

**Additional details and attachments**

If applicable, add any additional screenshots, logs, or other attachments to help explain your problem.

**Versions**
 - Deephaven: OSS
 - OS: Big Sur
 - Browser: Chrome
 - Docker: 20.10.6

**Comment by rcaudy** [comment]
I have done some testing of my own. I think the initially reported expectations don't match mine; I would expect that specifying an `int` value for a `byte` as in the provided example will result in the appropriate conversion (in the case, a "narrowing primitive conversion"). See https://docs.oracle.com/javase/specs/jls/se10/html/jls-5.html#jls-5.1.3 for details.

Here's another reproducer:
```
from deephaven.TableTools import newTable, byteCol, show, emptyTable, diff
t1=emptyTable(1).update("A=(byte)128")
t2=newTable(byteCol("A", 128))
show(t1)
show(t2)
print(diff(t1, t2,1))
```
This shows that the tables take on the values I would expect (`128` == `0x80` == `0b10000000` which is interpreted as `-127` as a byte). It also shows that these two mechanisms for producing the result do so identically. 

I see the same bad UI behavior as reported. The error therefore seems likely to lie in snapshotting, marshalling/unmarshalling, or display.

**Comment by rcaudy** [comment]
Problem is maybe unique to `byte`:

This renders fine:
```
t3=emptyTable(1).update("A=(short)0x8000")
show(t3)
         A
----------
(null)    
```
This renders fine, but shows the spinner:
```
t3=emptyTable(1).update("A=(byte)0x80")
show(t3)
         A
----------
(null)   
```

**Comment by niloc132** [comment]
This should be fixed now as part of #825.

**Comment by rcaudy** [comment]
This issue raised an important UI bug that Colin addressed. The larger ask (validate inputs to column construction methods) is not something we can accommodate; the handling of type conversions is a language-level issue. For Java, this is covered by the JLS.
